modules/gapi/include/opencv2/gapi/core.hpp

```diff
@@ -1757,6 +1861,71 @@ GAPI_EXPORTS GMat warpAffine(const GMat& src, const Mat& M, const Size& dsize, i
+Function can take GArray<Point2f>, GArray<Point2f> for 2D and 3D cases or GMat for any
+dimentionality
+
+@note In case of an N-dimentional points' set given, input Mat should be 2-dimensional, have
+a single row or column if there are N channels, or have N columns if there is a single channel.
+Mat should have @ref CV_32F depth
+
+@param K Number of clusters to split the set by.
+@param bestLabels Optional input integer array that can store the supposed initial cluster indices
+for every sample. Used when ( flags = #KMEANS_USE_INITIAL_LABELS ) flag is set.
+@param criteria The algorithm termination criteria, that is, the maximum number of iterations
+and/or the desired accuracy. The accuracy is specified as criteria.epsilon. As soon as each of
+the cluster centers moves by less than criteria.epsilon on some iteration, the algorithm stops.
+@param attempts Flag to specify the number of times the algorithm is executed using different
+initial labellings. The algorithm returns the labels that yield the best compactness (see the first
+function return value).
+@param flags Flag that can take values of cv::KmeansFlags
+
+@return Compactness measure that is computed as
+\f[\sum _i  \| \texttt{samples} _i -  \texttt{centers} _{ \texttt{labels} _i} \| ^2\f]
+after every attempt. The best (minimum) value is chosen and the corresponding labels and the
+compactness value are returned by the function.
+@return Integer array that stores the cluster indices for every sample.
+@return Array of the cluster centers.
+
+@note Basically, you can use only the core of the function: set the number of attempts to 1,
+initialize labels each time using a custom algorithm, pass them with the
+( flags = #KMEANS_USE_INITIAL_LABELS ) flag, and then choose the best (most-compact) clustering.
+*/
+GAPI_EXPORTS std::tuple<GOpaque<double>,GMat,GMat>
+kmeans(const GMat& data, const int K, const TermCriteria& criteria,
+       const int attempts, const KmeansFlags flags, const GMat& bestLabels);
+
+/** @overload
+@note Function textual ID is "org.opencv.core.kmeansNDNoInit"
+
+@note #KMEANS_USE_INITIAL_LABELS flag must not be set while using this overload.
+ */
+GAPI_EXPORTS std::tuple<GOpaque<double>,GMat,GMat>
+kmeans(const GMat& data, const int K, const TermCriteria& criteria, const int attempts,
+       const KmeansFlags flags);
+
+/** @overload
+@note Function textual ID is "org.opencv.core.kmeans2D"
+ */
+GAPI_EXPORTS std::tuple<GOpaque<double>,GArray<int>,GArray<Point2f>>
+kmeans(const GArray<Point2f>& data, const int K, const TermCriteria& criteria, const int attempts,
+       const KmeansFlags flags, const GArray<int>& bestLabels = GArray<int>(std::vector<int>{}));
+
+/** @overload
+@note Function textual ID is "org.opencv.core.kmeans3D"
+ */
+GAPI_EXPORTS std::tuple<GOpaque<double>,GArray<int>,GArray<Point3f>>
+kmeans(const GArray<Point3f>& data, const int K, const TermCriteria& criteria, const int attempts,
```

Comment:
Is Point3f necessary? I have looked to ocv documentation and this case isn't there.

Reply:
![image](https://user-images.githubusercontent.com/52420572/99798282-cef58e80-2b41-11eb-951d-cd358672e772.png)

Reply:
As you can see, all the for of these examples are for 2D points

Reply:
Ok. Does it work with Mat(,,CV_32FC3)? It is my interest.

Reply:
Maxim, thanks for the question! While checking your comment, found out that such cases are not fully supported. Fixed.